### PR TITLE
Point-based mechanism for compound attribute representation

### PR DESCRIPTION
**Pull request by kroenlein**
This is an alternate solution hypothesis for the multi-attribute/aggregation problem, as described in https://citrine.atlassian.net/wiki/spaces/PLA/pages/201949798/Multi-attribute+and+multi-point+aggregations.  Presently under review for clarity, usability and technical implications.

### CODE REVIEW

**Comment by lkubie** [comment]
I'm not sure I understand the point of this. In the examples given in the docs, why not have two Measurements. One being a vapor pressure Measurement with a Property of Vapor Pressure of 11 kPa and a Condition of temperature of 38 °C, and a second MEasurement being a Density measurement with a Property of density of 0.684 g/mL and a condition of Temperature of  25 °C? That is how I would tell Calcite to do this. I am not sure what pain point this is solving.

**Comment by kroenlein** [comment]
> I'm not sure I understand the point of this. In the examples given in the docs, why not have two Measurements. One being a vapor pressure Measurement with a Property of Vapor Pressure of 11 kPa and a Condition of temperature of 38 °C, and a second MEasurement being a Density measurement with a Property of density of 0.684 g/mL and a condition of Temperature of 25 °C? That is how I would tell Calcite to do this. I am not sure what pain point this is solving.

For the example given, this is a replacement for the PropertyAndConditions attached to the Material Spec, not Measurement Runs.  I bring up the SDS specificially because these are claims made by the manufacturer; our client would never perform this measurement; and it's likely that the supplier did not perform that measurement on this sample.

**Comment by bfolie** [comment]
Summarizing on offline conversation: I view this approach as being effectively identical to the multi-attribute solution, in that it's a way of grouping attributes (one attribute to one group, one group to many attributes). So the question is "what is more intuitive to users?" and "what integrates more smoothly into our existing stack?"

This seems to involve fewer breaking changes. This could be intuitive, but we need to think on the name. One option that makes sense to me is "cluster."

**Review by maxhutch — COMMENTED** [review]
The design looks good!

Request for clarification: if I have some attributes with `point=None` and others with `point="foo"`, are there two groups of attributes there or just one?  Basically, does `point=None` correspond to "match everything" or "default unique cluster"?

Request for examples: It would be good to see how this is used for:
 * PropertiesAndConditions (with conditions)
 * Curves
 * Repeated measurements

**Comment by kroenlein** [comment]
> Request for clarification: if I have some attributes with `point=None` and others with `point="foo"`, are there two groups of attributes there or just one? Basically, does `point=None` correspond to "match everything" or "default unique cluster"?

If an attribute has `point=None`, then it is affiliated with the Object and has not conflicting associations.  A barometer reading at the start of an experiment would have `point=None` and would be true over the course of any curves in the object.  So closer to "match everything".

**Comment by kroenlein** [comment]
> One option that makes sense to me is "cluster."

I like "cluster" but have gotten push-back from less technical folks.

**Comment by maxhutch** [comment]
> If an attribute has `point=None`, then it is affiliated with the Object and has not conflicting associations. A barometer reading at the start of an experiment would have `point=None` and would be true over the course of any curves in the object. So closer to "match everything".

That makes sense.  Could that explanation ^^ go in the text of the spec?

**Review by maxhutch — CHANGES_REQUESTED** [review]
Everything in here looks good.  There still don't seem to be any `cluster` examples in json.  It is set in one of the attribute examples, but only for a single attribute.  It would be really helpful to have an object example (a MaterialSpec, MeasurementRun, and/or ProcessRun) that actually demonstrates clustering of attributes into more than one cluster.

File: docs/specification/attributes.md [hunk 1]
```diff
@@ -143,27 +143,96 @@ len(`notes`)  | <=           | 32,768 (32KB), UTF-8 Encoded
 ```
 
 ```json
```
Comment (maxhutch) [inline docs/specification/attributes.md:145]:
```suggestion

**Two points on a temperature vs time trace:**

```json
```

**Review by maxhutch — APPROVED** [review]

**Pull request merged**